tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/modules/OSGiBundleListener.java

```diff
@@ -132,6 +133,17 @@ protected void handleTypedEvent( PropertyContentEvent event )
                     {
                     }
                 }
+
+                if( jar != null )
```

Comment:
For this to work we have to close the jar file before we call "new JarFile(file)" again or we lose the information about that jar file.  So we must close each "jarFile" before the next one is created or we need to remember each jar file that was opened and close them all after we are finished with it.